src/net/tcp/listener.rs

```diff
@@ -36,11 +40,17 @@ impl TcpListener {
             let host = world.current_host_mut();
 
             if !addr.ip().is_unspecified() && !addr.ip().is_loopback() {
-                panic!("{addr} is not supported");
```

Comment:
Can you keep the same message format for these? Having the arguments in the error message make it faster to debug the issue.